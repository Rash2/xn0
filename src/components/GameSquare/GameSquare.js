import React, {useEffect, useState} from "react";
import classes from "./GameSquare.module.css";

export const X = '💡';
export const O = '🏥';
export const Empty = '';
// turn 1 | 2
// 1 <=> X | 2 <=> O
const gameSquare = ({turn, onSymbolChange}) => {
  const [symbol, setSymbol] = useState(Empty);

  const onClick = () => {
    if (symbol === Empty) {
      if (turn === 1) {
        setSymbol(X);
      }
      if (turn === 2) {
        setSymbol(O);
      }
    }
  }

  useEffect(() => {
    if (typeof onSymbolChange === 'function') {
      onSymbolChange(symbol);
    }
  }, [symbol])

  return (
    <div data-testid='square' onClick={onClick} className={classes.GameSquare}>
      <p data-testid='square-text'>{symbol}</p>
    </div>
  );
};

export default gameSquare;
