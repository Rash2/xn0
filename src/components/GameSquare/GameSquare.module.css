.GameSquare {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100px;
    height: 100px;
    border: 1px solid black;
}

.GameSquare p {
    font-size: 1.5em;
}