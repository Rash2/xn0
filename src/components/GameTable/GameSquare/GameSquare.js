import React from "react";
import classes from "./GameSquare.module.css";

const gameSquare = ({ symbol, squareClicked, squareId }) => {
  return (
    <div onClick={() => symbol === '' ? squareClicked(squareId) : null} className={classes.GameSquare}>
      <p>{symbol}</p>
    </div>
  );
};

export default gameSquare;
