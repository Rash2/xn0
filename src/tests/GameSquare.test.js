import React from "react";
import renderer from 'react-test-renderer';
import GameSquare from "../components/GameTable/GameSquare/GameSquare";

const tempState = {gameState: ['','','','','','','','',''], currentSymbol: 'ðŸ’¡'};
// <GameSquare symbol={squareSymbol} squareClicked={squareClicked} squareId={index}/>

const squareClickedHandler = (squareIndex) => {
  setGameState(gameState.map((square, index) => index === squareIndex ? currentSymbol : square));
  toggleCurrentSymbolHandler();
}

describe("gameSquare", () => {
  test("square clicked", () => {
    const tree = renderer.create(<GameSquare symbol='ðŸ’¡' squareClicked={squareClickedHandler} squareId={0} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
