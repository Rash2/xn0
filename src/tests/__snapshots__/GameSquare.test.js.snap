// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gameSquare square clicked 1`] = `
<div
  onClick={[Function]}
>
  <p>
    ðŸ’¡
  </p>
</div>
`;
